app/lib/active_storage/previewer/csv_previewer.rb

```diff
@@ -0,0 +1,47 @@
+require "csv"
+require "chunky_png"
+
+module ActiveStorage
+  class Previewer::CSVPreviewer < Previewer
+    def self.accept?(blob)
+      blob.content_type == "text/csv"
+    end
+
+    def preview
+      download_blob_to_tempfile do |file|
+        rows = CSV.read(file.path).first(5)
+
+        png = render_table_as_image(rows)
+        tempfile = Tempfile.new(["preview", ".png"])
+        png.save(tempfile.path)
+
+        yield tempfile, format: :png
```

Comment:
Do we need to close the temp file? (maybe the `block` format will autoclose the file)